src/metabase/db/util.clj

```diff
@@ -66,3 +69,35 @@
   ;;    [:in :semantic_type #{"type/URL" "type/ImageURL"}]
   ([expr type-keyword]
    [:in expr (type-keyword->descendants type-keyword)]))
+
+(defmacro idempotent-insert!
```

Comment:
I'd note that `select-expr` shouldn't produce side-affects as it'll be evaluated multiple times